app/DoctrineMigrations/Version20220805151009.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Migrations;
+
+use AppBundle\Entity\Package;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20220805151009 extends AbstractMigration implements ContainerAwareInterface
+{
+    use ContainerAwareTrait;
+
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE package ADD slug VARCHAR(255)');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_DE686795989D9B62 ON package (slug)');
+    }
+
+    public function postUp(Schema $schema): void
+    {
+        $em = $this->container->get('doctrine.orm.entity_manager');
+
+        $packages = $this->container->get('doctrine')
+            ->getRepository(Package::class)
+            ->findAll();
+
+        foreach ($packages as $package) {
+            $package->setSlug($package->getName());
```

Comment:
Just to be sure: because you are using `postUp`, the name will be slugified automatically using the Doctrine behavior?

Reply:
Yes. There's a way to 'urilize' strings with this `Gedmo\Sluggable\Util\Urlizer::urlize(...)` but it doesn't take into account existing slugs in DB. So if you already have a slug 'small-box' in DB using Doctrine behaviour  for the same string it will generate the slug 'small-box-1', and etc.